src/lang/checkers/Cashflow.ml

```diff
@@ -1014,6 +1030,32 @@ module ScillaCashflowChecker
     let (new_p, _, new_ctr_tag_map, changes) = walk p ctr_tag_map in
     (new_p, new_ctr_tag_map, changes)
 
+  let update_pattern_vars_tags_from_scrutinee p scrutinee_tag =
+    let rec walk p expected_tag =
+      match p with
+      | Wildcard -> (Wildcard, false)
+      | Binder x ->
+          let new_x_tag = lub_tags expected_tag (get_id_tag x) in
+          let new_x = update_id_tag x new_x_tag in
+          (Binder new_x, new_x_tag <> (get_id_tag x))
+      | Constructor (s, ps) ->
+          let expected_subtags =
+            match ctr_pattern_to_subtags s expected_tag with
+            | Some ts -> ts
+            | None -> List.map ps ~f:(fun _ -> NoInfo) in
+          let new_subpatterns_with_tags =
```

Comment:
Just a suggestion, feel free to ignore it :)
```ocaml
let new_subpatterns_with_tags =
  List.zip ps expected_subtags |> Option.value ~default:[]
```

Reply:
Fixed.